### DISCUSSION TITLE
In c# scripting, is there a way to get to the Expression or FormatString for a dataset that is a Power BI dataset?

### DISCUSSION THREAD

**Discussion by dpesante** [TabularEditor/TabularEditor3#1148] [🙋♀️TE3 Q&A] [answered]
For example when I examine the measure properties using "m.Output();" I see the following in the FormatString property:

EXTERNALMEASURE(
    "_HMs With Active Assignments FormatString",
    STRING,
    "DirectQuery to AS - VEDA XRay - Template Skin - Datamodel"
)

Is there a way to take this string and run it so that it returns the actual FormatString of the measure in the Power BI Dataset?

**Comment by otykier** [comment, marked as answer]
Hi @dpesante 
You want to evaluate the DAX expression making up the FormatStringExpression, from within a C# script?

```csharp
var formatString = EvaluateDax(m.FormatStringExpression).ToString();
formatString.Output();
```

[`EvaluateDax`](https://docs.tabulareditor.com/api/TabularEditor.TOMWrapper.Database.html#TabularEditor_TOMWrapper_Database_EvaluateDax_System_String_)
